sdk/runanywhere-python/runanywhere/download.py

```
@@ -0,0 +1,524 @@
+
+# Dedup concurrent downloads to the SAME destination. Two resolve_model calls for one source
+# would otherwise open two write streams on the same `.part` file and corrupt it. A per-dest
+# state holder (minted under a lock) is the sync point: the first caller downloads; the rest WAIT
+# for it to finish (so they never hand back a ResolvedModel pointing at a not-yet-written file).
+_IN_FLIGHT_LOCK = threading.Lock()
+
+
+class _DownloadState:
+    """Shared outcome for one in-flight destination: an Event + the owner's error (if any)."""
+
+    __slots__ = ("event", "error")
```

Ruff (0.15.21)

[warning] `_DownloadState.__slots__` is not sorted

Apply a natural sort to `_DownloadState.__slots__`

(RUF023)